### ISSUE TITLE
callout-note-title localization not working as expected

### ISSUE THREAD

**Issue by MarcoPortmann** [quarto-dev/quarto-cli#8281]
### Bug description

I would like to cross-reference a callout note in a German Quarto document. Therefore, the reference should have the prefix "Hinweis" instead of note. As I understand it, this should work with

```markdown
language:
  de:
    callout-note-title: "Hinweis"
``` 

in the yaml header.  The reference is still labeled "Note 1". Am I doing something wrong? 

### Steps to reproduce

````markdown
---
title: "Reproducible Quarto Document"
format: html
lang: de
language:
  de:
    callout-note-title: "Hinweis"
---
  
# A title  
This is a reproducible Quarto document.


::: {#nte-test .callout-note collapse="true" appearance="simple"}
## This is a callout note

Some text.
:::

This is @nte-test.


The end.

````

### Expected behavior

The reference should be labeled "Hinweis 1".

### Actual behavior

The reference is still labeled "Note 1".

### Your environment

RStudio Server
2023.09.0 Build 463

### Quarto check output

1.4.542

**Comment by mcanouil** [comment]
Thanks for the report!

FYI, no need to set:

```yaml
language:
  de:
    callout-note-title: "Hinweis"
 ```
 
 It's the default.
 
 - https://github.com/quarto-dev/quarto-cli/blob/cfeb822ca69e14f8d24bba57bc501b7ba8e2c8e6/src/resources/language/_language-de.yml#L38

**Comment by MarcoPortmann** [comment]
Two questions to make sure I have understood correctly:

There is a bug in that the default value from _language-en.yml is not applied as well?
And if I wanted to set a different title
```markdown
language:
  de:
    callout-note-title: "Vertiefungsbox"
```
would the correct way to do it, right?

**Comment by cscheid** [comment]
> There is a bug in that the default value from _language-en.yml is not applied as well

Yes, that's correct. What's happening here is that the crossreferenceable title resolution happens against the `crossref-{KEY}-title` metadata entry, while the callout title resolution happens against the `callout-{TYPE}-title` metadata entry.

As a current workaround, you should be able to provide the correct localization with

```yml
language:
  de:
    crossref-nte-title: "Hinweis"
    crossref-nte-prefix: "Hinweis"
```

A couple of things to note: for _callouts_, you write "callout-**`note`**-title". For _crossreferenceable callouts_ you write "crossref-**`nte`**-title", and you also have to write `crossref-nte-prefix`, because in the case of crossreferenceable entries, these can be written differently.

This is a bit unfortunate, but inevitable for the moment. We'll consider making these uniform in a later Quarto version.

**Comment by MarcoPortmann** [comment]
Thank you for this workaround. This works fine for `format: html`. I've tested the same for pdf.

```markdown
---
title: "Reproducible Quarto Document"
format: pdf
lang: de
language:
  de:
    crossref-nte-title: "Hinweis"
    crossref-nte-prefix: "Hinweis"
---
  
::: {#nte-test .callout-note icon=false}
## Test

Text.
:::

This is @nte-test.
```

This yields:
![image](https://github.com/quarto-dev/quarto-cli/assets/7468121/4d88b56c-9981-45c7-b10c-dec4657e9df3)

Does `format: pdf` even require another localized expression for the title?

**Comment by cscheid** [comment]
> Does format: pdf even require another localized expression for the title?

No, it shouldn't. 

It's really that the bug workaround doesn't appear to work in `pdf`.

**Comment by MarcoPortmann** [comment]
Ok, thanks for the clarification.

**Comment by mcanouil** [comment]
For reference, the issue was also reported later on <https://github.com/quarto-dev/quarto-cli/issues/9263>.

**Comment by cscheid** [comment]
@cderv Do you still want to take a shot at this one?

**Comment by MarcoPortmann** [comment]
We would be very grateful for a fix. I currently have pdfs where I use adobe acrobat pro to manually correct the callout headings.

**Comment by Noe-J-Nava** [comment]
Is there any development on this issue?

**Comment by mcanouil** [comment]
@Noe-J-Nava The team updates issues when there is something to update about. Instead of asking, you could simply use the reaction button on the original post to show that you are interested to see this fixed.
